app/models/host_facet_base.rb

```diff
@@ -0,0 +1,52 @@
+class HostFacetBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  def self.inherited(subclass)
+    super
+    subclass.class_eval do
+      belongs_to_host
+    end
+  end
+
+  # Add facet's details to host's ENC by returning it from this method
+  # Basic ENC hash structure:
+  # ---
+  # classes:
+  #   ...
+  # parameters:
+  #   ...
+  # environment:
+  #   ...
+  def info
+    {}
+  end
+
+  # Specify any smart proxy id's used by this facet.
+  def smart_proxy_ids
+    []
+  end
+
+  # This method will be called after a host with all its facets is cloned.
+  def after_clone
```

Comment:
If it's not used up there let's remove the endpoint in the generic facet as well